### PR TITLE
GC+: Add exceptions and error handling

### LINKED ISSUE #4803: GC+: Add exceptions and error handling

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #4803

### CODE REVIEW

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala [hunk 1]
```diff
@@ -78,21 +78,21 @@ object UncommittedGarbageCollector {
...
-      Console.out.printf("Please provide a mark ID (%s) for sweep-only mode. Exiting...\n",
-                         LAKEFS_CONF_GC_MARK_ID
-                        )
-      System.exit(2)
+      throw new ParameterValidationException(
+        s"Please provide a mark ID ($LAKEFS_CONF_GC_MARK_ID) for sweep-only mode. Exiting...\n"
+      )
     } else if (shouldMark && markID.nonEmpty) {
-      Console.out.println("Can't provide mark ID for mark mode. Exiting...")
+      throw new ParameterValidationException("Can't provide mark ID for mark mode. Exiting...")
```
Comment (nopcoder) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:95]: can remove the elses

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala [hunk 2]
```diff
@@ -237,22 +228,25 @@ object UncommittedGarbageCollector {
     cachedAddresses.write.text(s"$reportDst/deleted.text")
   }
 
-  private def reportPath(storageNamespace: String, runID: String): String = {
+  private def runPath(storageNamespace: String, runID: String): String = {
```
Comment (nopcoder) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:231]: Can we name it 'formatRunPath' or something that we don't think we are actually running something.

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala [hunk 3]
```diff
@@ -199,11 +195,6 @@ object UncommittedGarbageCollector {
           .toDF()
         removed.collect()
       }
-    } catch {
-      case e: Throwable =>
-        success = false
```
Comment (nopcoder) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:204]: Nice - just initialize `success` with false and set it to true at the end of the 'try' scope.

**Review by nopcoder — CHANGES_REQUESTED** [review]
Looks good - minor comments, my main concern before we make this one is the `lakefs.gc.do_mark=true` as it will change the behaviour we just documented does not align with the current GC process.

**Comment by N-o-Z** [comment]
> Looks good - minor comments, my main concern before we make this one is the `lakefs.gc.do_mark=true` as it will change the behaviour we just documented does not align with the current GC process.

Aligned defaults with committed garbage collector for now.
Modified the documentation (which had a bug in it as well)

**Review by nopcoder — APPROVED** [review]
LGTM - thanks!

**Pull request merged**